docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +326,217 @@ recursion was explicitly requested.
+### 22.1.13. `{:timeLimitMultiplier X}`
 This attribute may be placed on a method or function declaration
 and has an integer argument. If `{:timeLimitMultiplier X}` was
 specified a `{:timelimit Y}` attributed is passed on to Boogie
 where `Y` is `X` times either the default verification time limit
 for a function or method, or times the value specified by the
 Boogie `timelimit` command-line option.
 
-### 22.1.16. trigger
+### 22.1.14. `{:trigger}` {#sec-trigger}
 Trigger attributes are used on quantifiers and comprehensions.
 They are translated into Boogie triggers.
 
-### 22.1.17. typeQuantifier
+### 22.1.15. `{:typeQuantifier}`
 The `{:typeQuantifier}` attribute must be used on a quantifier if it
 quantifies over types.
 
 
-## 22.2. Boogie Attributes
-Use the Boogie "/attrHelp" option to get the list of attributes
-that Boogie recognizes and their meaning. Here is the output at
-the time of this writing. Dafny passes attributes that have
-been specified to Boogie.
+## 22.2. Boogie-specific verification attributes
+Dafny passes verification attributes that have been specified to its dependency Boogie.
+Use the `/attrHelp` option to get the list of attributes of both Dafny and Boogie
+recognized by Boogie and their meaning.
 
+### 22.2.1. Verification attributes on top-level declarations
+
+#### 22.2.1.1. `{:ignore}`
+Ignore the declaration (after checking for duplicate names).
+
+#### 22.2.1.2. `{:extern <name>}` {#sec-extern}
+
+If a ``ClassDecl``, a ``MethodDecl`` or a ``FunctionDecl`` has an `{:extern}` attribute,
+then:
+* The compiler generally emits nothing for it and will use <name> instead of the dafny provided name in the generated code referencing it.
+* `{:extern}` Methods and functions can have [`requires` clauses](#sec-requires-clause) and [`ensures` clauses](#sec-ensures-clause).
+* `{:extern}` methods typically do not have any body, since they are unchecked and opaque.
+* `{:extern}` Functions can have a body which is used only for verification purposes.
+
+The difference with [`{:axiom}`](#sec-axiom) is that the compiler will still emit code for an [`{:axiom}`](#sec-axiom), if it is a [`function method`, a `method` or a `function by method`](#sec-function-declarations) with a body.
+
+#### 22.2.1.3. `{:checksum <string>}`
+
+Attach a checksum to be used for verification result caching.
+
+### 22.2.2. Verification attributes on functions and methods
+
+#### 22.2.2.1. `{:inline N}`
+     
+Inline a given procedure (can be also used on implementation).
+N should be a non-negative number and represents the inlining depth.
+With `/inline:assume`, call is replaced with `assume false` once inlining depth is reached.
+With `/inline:assert`, call is replaced with `assert false` once inlining depth is reached.
+With `/inline:spec`, call is left as is once inlining depth is reached.
+With the above three options, methods with the attribute `{:inline N}` are not verified.
+With `/inline:none` the entire attribute is ignored.
```

Comment:
Boogie's `:inline` attribute expects a called `procedure P` to having a matching `implementation P`. That's not how Dafny generates Boogie code, so `:inline` has no effect on Dafny programs. I suggest removing this section. Alternatively, replace the contents of this section to say that Boogie's `:inline` attribute has no effect on Dafny programs.

Reply:
I removed this section altogether. Boogie documentation should stay within Boogie, unless it's useful in Dafny.